tfx/examples/cifar10/meta_data_writer.py

```diff
@@ -0,0 +1,208 @@
+    input_normalization = _metadata_fb.ProcessUnitT()
+    input_normalization.optionsType = (
+        _metadata_fb.ProcessUnitOptions.NormalizationOptions)
+    input_normalization.options = _metadata_fb.NormalizationOptionsT()
+    input_normalization.options.mean = self.model_info.mean
+    input_normalization.options.std = self.model_info.std
+    input_meta.processUnits = [input_normalization]
+    input_stats = _metadata_fb.StatsT()
+    input_stats.max = [self.model_info.image_max]
+    input_stats.min = [self.model_info.image_min]
+    input_meta.stats = input_stats
+
+    # Creates output info.
+    output_meta = _metadata_fb.TensorMetadataT()
+    output_meta.name = "probability"
+    output_meta.description = "Probabilities of the %d labels respectively." % \
+                                self.model_info.num_classes
+    output_meta.content = _metadata_fb.ContentT()
+    output_meta.content.content_properties = _metadata_fb.FeaturePropertiesT()
+    output_meta.content.contentPropertiesType = (
+        _metadata_fb.ContentProperties.FeatureProperties)
+    output_stats = _metadata_fb.StatsT()
+    output_stats.max = [1.0]
+    output_stats.min = [0.0]
+    output_meta.stats = output_stats
+    label_file = _metadata_fb.AssociatedFileT()
+    label_file.name = os.path.basename(self.label_file_path)
+    label_file.description = "Labels for objects that the model can recognize."
+    label_file.type = _metadata_fb.AssociatedFileType.TENSOR_AXIS_LABELS
+    output_meta.associatedFiles = [label_file]
+
+    # Creates subgraph info.
+    subgraph = _metadata_fb.SubGraphMetadataT()
+    subgraph.inputTensorMetadata = [input_meta]
+    subgraph.outputTensorMetadata = [output_meta]
+    model_meta.subgraphMetadata = [subgraph]
+
+    b = flatbuffers.Builder(0)
+    b.Finish(
+        model_meta.Pack(b),
+        _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER)
+    self.metadata_buf = b.Output()
+
+  def _populate_metadata(self):
+    """Populates metadata and label file to the model file."""
+    populator = _metadata.MetadataPopulator.with_model_file(self.model_file)
+    populator.load_metadata_buffer(self.metadata_buf)
+    populator.load_associated_files([self.label_file_path])
+    populator.populate()
+
+
+def main(_):
+  model_file = FLAGS.model_file
+  model_basename = os.path.basename(model_file)
+  if model_basename not in _MODEL_INFO:
+    raise ValueError(
+        "The model info for, {0}, is not defined yet.".format(model_basename))
+
+  export_model_path = os.path.join(FLAGS.export_directory, model_basename)
+
+  # Copies model_file to export_path.
+  tf.io.gfile.copy(model_file, export_model_path, overwrite=True)
+
+  # Generate the metadata objects and put them in the model file
+  populator = MetadataPopulatorForImageClassifier(
+      export_model_path, _MODEL_INFO.get(model_basename), FLAGS.label_file)
+  populator.populate()
+
+  # Validate the output model file by reading the metadata and produce
```

Comment:
Is the json file useful? If so, why is it "validation"? If not, then perhaps have a validation flag and perhaps store the json file in tmpdir instead of placing it with the export.

Reply:
The json file serves as a human-readable file to tell user what their model's metadata looks like. I think it has some value for the user. It's "validation" in a sense that it verify the exported model's metadata can be read in properly. I can change the comment if you think it's not very appropriate.

Reply:
how do you feel about having a commandline flag of the form: also_produce_json_metadata?

Reply:
sure sounds good